npc/re/custom/lasagna/lasagna_npcs.txt

```diff
@@ -2463,44 +2463,53 @@ lasagna,140,245,3	script	The sensitive-minded ela	4_CAT_SAILOR3,{
 	mes "Yota often makes quality amulets, but he still needs more polishing.";
 	mes "If he pays attention to details, his amulets will be even greater. But he just doesn't care.";
 	next;
-	.@s = (select( "Polish the Horse Mackerel Amulet.", "Polish the Leaf Amulet.", "Polish the Rabbit Amulet." ) - 1) * 4;
-
-	setarray .@item_data[0],
-		28413, 30,	// Basic Saurel Charm [1]
-		28414, 90,	// Fair Saurel Charm [1]
-		28416, 30,	// Basic Leaf Charm [1]
-		28417, 90,	// Fair Leaf Charm [1]
-		28419, 30,	// Basic Bunny Charm [1]
-		28420, 90;	// Fair Bunny Charm [1]
-
-	if ((countitem(.@item_data[.@s]) - isequippedcnt(.@item_data[.@s]))  > 0)
-		.@index = .@s;
-	else if ((countitem(.@item_data[.@s+2]) - isequippedcnt(.@item_data[.@s+2])) > 0)
-		.@index = .@s + 2;
-	else {
+	.@s = select( "Polish the Horse Mackerel Amulet.", "Polish the Leaf Amulet.", "Polish the Rabbit Amulet." ) - 1;
+	setarray .@ids[0], 28413,28416,28419;
+	.@start_ID = .@ids[.@s];
+	if ((countitem(.@start_ID) - isequippedcnt(.@start_ID))  > 0) {
+		.@your_charm_ID = .@start_ID;
+		.@next_charm_ID = .@start_ID + 1;
+		.@needCoin = 30;
+	}
+	else if ((countitem(.@start_ID+1) - isequippedcnt(.@start_ID+1))  > 0) {
```

Comment:
Maybe add a small comment what this +1 and +2 below are doing. Got me really confused.